### PR TITLE
Scanner criteria improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/funTest/kotlin/experimental/AbstractProvenanceBasedStorageFunTest.kt [hunk 1]
```diff
@@ -188,14 +190,6 @@ abstract class AbstractProvenanceBasedStorageFunTest(vararg listeners: TestListe
     }
 }
 
-private fun ScannerDetails.toCriteria() =
```
Comment (oheger-bosch) [inline scanner/src/funTest/kotlin/experimental/AbstractProvenanceBasedStorageFunTest.kt:191]: You have changed the `configMatcher` in your consolidated version: Some tests use a matcher that always returns *true*, while others set an exact matcher. This is probably the reason why multiple tests are failing. You might have to make the matcher configurable in the `ScannerCriteria.forDetails()`.

**Review by oheger-bosch — DISMISSED** [review]

**Comment by sschuberth** [comment]
Note that this doe snot yet implement the idea to make a `ScannerCriertia`'s `maxVersion` inclusive. These are just cleanups and improvements to the current state.

**Review by mnonnenmacher — APPROVED** [review]

File: scanner/src/funTest/kotlin/experimental/AbstractProvenanceBasedStorageFunTest.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/funTest/kotlin/experimental/AbstractProvenanceBasedStorageFunTest.kt:191]: The error was actually in different `maxVersion`s used; I've fixed that.

**Comment by sschuberth** [comment]
Merging despite the unrelated `BabelFunTest` failure.

**Pull request merged**